packages/clerk-js/src/core/jwt-client.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  ClientJSON,
+  OrganizationMembershipJSON,
+  PartialWithClerkResource,
+  PublicUserDataJSON,
+  SessionJSON,
+  TokenJSON,
+  UserJSON,
+} from '@clerk/types';
+
+import { Token } from './resources';
+import { Client } from './resources/Client';
+
+/**
+ * Create a new client instance from a jwt.
+ * The caller is responsible for reading the jwt from the `__session` cookie.
+ */
+export function createClientFromJwt(jwt: string | undefined | null): Client {
+  // Use `Token` class to parse the JWT token
+  let token;
```

Comment:
```suggestion
  let token: Token | null;
```

Can we type it earlier rather than leaving the comment about it?